src/Tracing/Instrumentation/EventSubscriber/AddUserEventSubscriber.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the worldia/instrumentation-bundle package.
+ * (c) Worldia <developers@worldia.com>
+ */
+
+namespace Instrumentation\Tracing\Instrumentation\EventSubscriber;
+
+use Instrumentation\Tracing\Instrumentation\MainSpanContext;
+use Instrumentation\Tracing\Instrumentation\TracerAwareTrait;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class AddUserEventSubscriber implements EventSubscriberInterface
+{
+    use TracerAwareTrait;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => [['onRequestEvent', 0]],
```

Comment:
It might be interesting to give it a higher priority so that it's register before listeners with the default priority.
The firewall listener has a priority of `8` so as long as we are after this one it should be good.

Reply:
Are you suggesting a priority of 7?

Reply:
At least between 1 and 7 (inclusive).
But yes 7 seems OK, I would say as early as possible.

But I'm not sure about how the span is actually registered, maybe it does not matter if the tracer is designed as a unit of work.